File: eyan_helper/src/test/java/eu/eyan/testutil/ExcelAssert.java
package eu.eyan.testutil;

import static org.fest.assertions.Assertions.assertThat;

import java.io.File;
import java.io.IOException;

import jxl.Workbook;
import jxl.WorkbookSettings;
import jxl.read.biff.BiffException;

public class ExcelAssert {

	public static void assertExcelCell(File sourceFile, String sheetName, int column, int row, String expected) {
		Workbook workbook = null;
		try {
			workbook = Workbook.getWorkbook(sourceFile, getWorkbookSettings());
			assertThat(workbook.getSheet(sheetName).getCell(column - 1, row - 1).getContents()).isEqualTo(expected);
		} catch (BiffException | IOException e) {
			e.printStackTrace();
		} finally {
			if (workbook != null) {
				workbook.close();
			}
		}

	}

	public static WorkbookSettings getWorkbookSettings() {
		WorkbookSettings ws = new WorkbookSettings();
		ws.setEncoding("Cp1252");
		return ws;
	}
}

File: eyan_helper/src/main/scala/eu/eyan/testutil/ExcelAssert.scala
package eu.eyan.testutil

import java.io.File
import java.io.IOException

import org.fest.assertions.Assertions.assertThat

import jxl.Workbook
import jxl.WorkbookSettings
import jxl.read.biff.BiffException

object ExcelAssert {
  def getWorkbookSettings = {
    val ws = new WorkbookSettings
    ws.setEncoding("Cp1252")
    ws
  }

  def assertExcelCell(sourceFile: File, sheetName: String, column: Int, row: Int, expected: String) = {
    var workbook: Workbook = null
    try {
      workbook = Workbook.getWorkbook(sourceFile, getWorkbookSettings)
      assertThat(workbook.getSheet(sheetName).getCell(column - 1, row - 1).getContents()).isEqualTo(expected)
    }
    catch {
      case e: BiffException => e.printStackTrace
      case e: IOException   => e.printStackTrace
    }
    finally {
      if (workbook != null) workbook.close 
    }
  }
}